### PR TITLE
Add unauthorized page (resolves #54)

### LINKED ISSUE #54: Create Unauthorized Page
If a user attempts to log in to Boardwalk and is rejected because she/he is not authorized, we should redirect to a page with details.

The page should say something like:

We are sorry but this account <user email> is not authorized for this access to this project <project>. If you believe you are using the correct email address and should be authorized to access this project please contact <contact email>.

Where

1. <user email> is the email of the user who logged in, but got rejected
1. <project> is the name of the project, e.g., Stage, Common, AnVIL, etc.
1. <contact email> is an email address the user can contact to request access.

<project> and <contact email> should be configurable, driven from environment variables set up via install_bootstrap in cgp_deployment repo.

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/DASH-22)
┆Project Name: CGP-Dashboard
┆Issue Number: DASH-22

### PR DESCRIPTION
**Pull request by coverbeck**
Example page:

screen shot 2018-08-27 at 3 29 22 pm

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -377,6 +379,14 @@ def boardwalk():
 def privacy():
     return redirect(url_for('privacy'))
 
+@app.route('/unauthorized')
+def unauthorized():
+    account = request.args.get('account', default=None)
+    project = os.getenv('PROJECT_NAME', '')
+    contact = os.getenv('CONTACT_EMAIL', '')
```
Comment (jessebrennan) [inline app.py]: maybe a slightly more sensible default could be used? like 'your administrator' for `contact`. Just an idea

File: app.py [hunk 2]
```diff
@@ -434,6 +444,9 @@ def callback():
         if resp.status_code == 200:
             user_data = resp.json()
             email = user_data['email']
+            # If so configured, check for whitelist and redirect to
+            # unauthorized page if not in whitelist, e.g.,
+            # return redirect(url_for('unauthorized', account=<redacted email>)),
```
Comment (jessebrennan) [inline app.py:449]: Thanks for putting that in there! This makes it easy for me ;)

File: app.py (hunk 1 above)
Comment (jessebrennan) [reply app.py]: On second thought, I think these values have to be set so using a default wouldn't really do anything.

**Review by jessebrennan — APPROVED** [review]
This looks great! I like it a lot. 

I haven't tried deploying it, but I'm assuming you have. Anyway that's something that should be tested before a merge.

Also It would be nice if the styling could happen in a CSS file instead of within the HTML to make it more reusable. probably not all that important though, especially since this is small scale.

**Comment by coverbeck** [comment]
1. The screenshot is from running Flask locally. It's not the full blown deploy, but it should be good. But good point, I'll try with my VM as well to be safe.
2. Regarding the CSS, I was thinking I just wanted a file that would load quickly without any dependencies. I also came up with this UX "design" on my own, that's somewhat inspired by one I worked on in the past. For better or worse, it's not a pattern that we use on other pages, because this page is not like any other pages, so far. All that said, it probably is best to put the the CSS in a file.

File: app.py (hunk 2 above)
Comment (coverbeck) [reply app.py:449]: We should discuss the `redacted email` when you get to it. I was thinking something like `coverbeck@***.com` or `coverbeck@...edu`, `coverbeck@...m`, etc.

**Comment by jessebrennan** [comment]
>1. The screenshot is from running Flask locally. It's not the full blown deploy, but it should be good. But good point, I'll try with my VM as well to be safe.

Sounds good!

>2. Regarding the CSS, I was thinking I just wanted a file that would load quickly without any dependencies. I also came up with this UX "design" on my own, that's somewhat inspired by one I worked on in the past. For better or worse, it's not a pattern that we use on other pages, because this page is not like any other pages, so far. All that said, it probably is best to put the the CSS in a file.

Up to you really, I just though I'd bring it up.

File: app.py (hunk 1 above)
Comment (coverbeck) [reply app.py]: I think I will remove the defaults, and if the value ends up being `None`, it will not display anything on the HTML side.

**Comment by coverbeck** [comment]
Now leaning towards not separating out the CSS. :) It doesn't really fit in any of the existing CSS files, and it currently won't be reused anyway. We can cross the bridge later if we need to reuse it.

**Comment by coverbeck** [comment]
@jessebrennan I ended up:

1. Keeping the CSS in the HTML
2. Leaving the defaults for project and contact as empty string, but not account, because I explicitly test for `None` in the template.

Could you please take one final look?

Also, I'm thinking it would be nice to have a contact name and a contact email, that way we wouldn't display the email. But that will require an update to the installer, and is nice to have, so we can tackle that later.

**Review by jessebrennan — APPROVED** [review]

**Pull request merged**